core-backend/common/src/lib.rs

```diff
@@ -70,16 +67,28 @@ pub enum TerminationReason {
     Exit(ProgramId),
     Leave,
     Success,
-    #[from]
-    Trap(TrapExplanation),
     Wait(Option<u32>, MessageWaitedType),
     GasAllowanceExceeded,
+
+    #[from]
+    Trap(TrapExplanation),
 }
 
 #[derive(
-    Decode, Encode, TypeInfo, Debug, Clone, PartialEq, Eq, PartialOrd, Ord, derive_more::Display,
+    Decode,
+    Encode,
+    TypeInfo,
+    Debug,
+    Clone,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    derive_more::Display,
+    derive_more::From,
 )]
 pub enum TrapExplanation {
+    #[from]
```

Comment:
This way we'll also allow `From<TrimmedString>`
```suggestion
```

Reply:
Done